public/app/features/plugins/admin/pages/PluginDetails.tsx

```diff
@@ -10,43 +11,51 @@ import { PluginDetailsHeader } from '../components/PluginDetailsHeader';
 import { PluginDetailsBody } from '../components/PluginDetailsBody';
 import { Page as PluginPage } from '../components/Page';
 import { Loader } from '../components/Loader';
-import { PluginTabLabels, PluginDetailsTab } from '../types';
-import { useGetSingle, useFetchStatus } from '../state/hooks';
+import { PluginTabLabels, PluginTabIds, PluginDetailsTab } from '../types';
+import { useGetSingle, useFetchStatus, useFetchDetailsStatus } from '../state/hooks';
 import { usePluginDetailsTabs } from '../hooks/usePluginDetailsTabs';
 import { AppNotificationSeverity } from 'app/types';
 import { PluginDetailsDisabledError } from '../components/PluginDetailsDisabledError';
 
 type Props = GrafanaRouteComponentProps<{ pluginId?: string }>;
 
-type State = {
-  tabs: PluginDetailsTab[];
-  activeTabIndex: number;
-};
-
-const DefaultState = {
-  tabs: [{ label: PluginTabLabels.OVERVIEW }, { label: PluginTabLabels.VERSIONS }],
-  activeTabIndex: 0,
-};
-
-export default function PluginDetails({ match }: Props): JSX.Element | null {
-  const { pluginId = '' } = match.params;
-  const [state, setState] = useState<State>(DefaultState);
+export default function PluginDetails({ match, queryParams }: Props): JSX.Element | null {
+  const {
+    params: { pluginId = '' },
+    url,
+  } = match;
+  const pageId = (queryParams.page as PluginTabIds) || PluginTabIds.OVERVIEW;
+  const parentUrl = url.substring(0, url.lastIndexOf('/'));
+  const defaultTabs: PluginDetailsTab[] = [
+    {
+      label: PluginTabLabels.OVERVIEW,
+      icon: 'file-alt',
+      id: PluginTabIds.OVERVIEW,
+      href: `${url}?page=${PluginTabIds.OVERVIEW}`,
+    },
+    {
+      label: PluginTabLabels.VERSIONS,
+      icon: 'history',
+      id: PluginTabIds.VERSIONS,
+      href: `${url}?page=${PluginTabIds.VERSIONS}`,
+    },
+  ];
+  const [activeTabIndex, setActiveTabIndex] = useState(Object.values(PluginTabIds).indexOf(pageId));
   const plugin = useGetSingle(pluginId); // fetches the localplugin settings
-  const { tabs } = usePluginDetailsTabs(plugin, DefaultState.tabs);
-  const { activeTabIndex } = state;
-  const { isLoading } = useFetchStatus();
+  const { tabs } = usePluginDetailsTabs(plugin, defaultTabs);
+  const { isLoading: isFetchLoading } = useFetchStatus();
+  const { isLoading: isFetchDetailsLoading } = useFetchDetailsStatus();
   const styles = useStyles2(getStyles);
-  const setActiveTab = useCallback((activeTabIndex: number) => setState({ ...state, activeTabIndex }), [state]);
-  const parentUrl = match.url.substring(0, match.url.lastIndexOf('/'));
 
   // If an app plugin is uninstalled we need to reset the active tab when the config / dashboards tabs are removed.
   useEffect(() => {
     if (activeTabIndex > tabs.length - 1) {
-      setActiveTab(0);
+      setActiveTabIndex(0);
```

Comment:
I've kept this approach of tracking the `activeTabIndex` for the time being but it prevents us from linking directly to a config or dashboard tab. I plan to revisit it in a follow up PR.

Reply:
Have opted for a slightly different approach.

- Changed the useEffect for solving `uninstall an app plugin and user is viewing a config / dashboard tab` which allows us to deep link to PluginDetails pages by comparing previous and current tabs length.
- Decided to add a page not found notice for `navigate to an unknown tab (e.g page=sausages or page=config but the app plugin isn't installed)`.